src/common.rs

```diff
@@ -112,6 +112,23 @@ pub fn do_bind(host: &str) -> Result<(RawFd, Domain)> {
     Ok((fd, domain))
 }
 
+/// Creates a unix socket for client.
+pub(crate) unsafe fn client_connect_unix(path: &str) -> Result<RawFd> {
+    let fd = socket(
```

Comment:
First, the socket  should be set SOCK_CLOEXEC to prevent the fd leak into child process, second here the unix path only supported the abstract unix socket, I think we should define a protocol here to indicate the unix socket type: abstract unix socket or path unix socket. Such as: the unix://<path> represented as an path unix socket, and the simple <path> means the abstract unix socket?

Reply:
I've refactored `do_bind` to create sockets for both client and server.
This should keep the same logic for both parties.